### DISCUSSION TITLE
Upgrade to a more recent version of sonarqube API

### DISCUSSION THREAD

**Discussion by christopheroger** [spotbugs/sonar-findbugs#964] [General]
Hello team, 
are there plans to to update the Sonarqube API to a more recent version ? I know it is done in the sq-10 branch. I tried it last year and it seemed to work well. But the branch is still not merged on the main branch.

Thanks.

**Comment by gtoison** [comment]
Hello,
May I ask why you're interested in upgrading the version? Is there some feature you're missing?
The sq-10 branch was created when it seemed that the way we are loading rules would be removed from the API, but so far it still exists. Due to the limited resources (time...) on my end, that branch isn't merged yet.
The integration tests of the project actually compile and test against various versions of the sonar API (through 10.3) so we're at least checking that we are compatible with recent versions

**Reply by christopheroger** [reply]
Ok, thank you for the answer.
And yes, for our use cases we need some new features related to security rules and they are only possible with a newer API:
- some findsecbugs rules generate too many false positives but are useful in certain cases. I need them to be categorized as "security hotspots" rather than issues
- When possible, security rules should be categorized in owasps top tens from sonarqube (at least top ten 2021). 

I started some developments last year from the sq-10 branch but I was waiting for a merge to go further

**Reply by gtoison** [reply]
I see, let me try to refresh my memory on that sq-10 branch

**Reply by gtoison** [reply]
I've merged the recent changes into the sq-10 branch
It would be much appreciated if you could test it and report any issue. The integration tests are running fine so it should be OK for the most part

**Reply by christopheroger** [reply]
Thanks for the merge, I will test the updated branch

**Comment by hazendaz** [comment]
I believe this issue has since been addressed and no longer needs open, if not the case please open new discussion.
